org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/messages.properties

```diff
@@ -1176,11 +1175,12 @@
 2000 = As of release 22, '_' is only allowed to declare unnamed patterns, local variables, exception parameters or lambda parameters
 2001 = Unnamed variables must have an initializer
 
+2011 = Comparing primitive wrapper types or Strings using '=='
```

Comment:
Well, it compares references instead of values, which is the problem 😅 

How about:
- `Unlikely argument type for '==': java.lang.Byte`
- `Unlikely argument type for '==': java.lang.String`
- `Unlikely argument type for '==': java.lang.Long`

?

Reply:
Essentially agree, perhaps:
* `Unlikely operand type for '==': java.lang.Long is a reference type`

Next I wondered if indeed both sides have the same type by definition, which allowed me to trick your analysis:

```
		Long l = 13l;
		Number i = Integer.valueOf(13);
		if (i == l)
			System.out.println("equal");
```

We should (IMO) report, but it keeps silent.

Changing the above snippet to `Object i` would be borderline. Should we report comparison wrapper vs. Object? String vs. Object? String vs. CharSequence? Serializable?

Reply:
> `Unlikely operand type for '==': java.lang.Long is a reference type`

plus we need at least one test for '!=', which would reveal that the message is not yet correct for that case :)

Reply:
Added.

Reply:
> Essentially agree, perhaps:
> 
> * `Unlikely operand type for '==': java.lang.Long is a reference type`
> 
> Next I wondered if indeed both sides have the same type by definition, which allowed me to trick your analysis:
> 
> ```
> 		Long l = 13l;
> 		Number i = Integer.valueOf(13);
> 		if (i == l)
> 			System.out.println("equal");
> ```
> 
> We should (IMO) report, but it keeps silent.
> 
> Changing the above snippet to `Object i` would be borderline. Should we report comparison wrapper vs. Object? String vs. Object? String vs. CharSequence? Serializable?

I've implemented Number and reference comparisons. I doubt we should also do String vs CharSequence. We might consider String vs references as well, such as implemented for Number?